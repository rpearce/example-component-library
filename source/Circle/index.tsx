import React, { FC } from 'react'

interface Props {
  desc?: string
  size: number
  title?: string
}

const Circle: FC<Props> = ({ desc, size, title }) => (
  <svg
    className="rl-circle"
    height={size}
    role="img"
    viewBox="0 0 100 100"
    width={size}
    xmlns="http://www.w3.org/2000/svg"
  >
    {title && <title>{title}</title>}
    {desc && <desc>{desc}</desc>}
    <circle cx="50" cy="50" r="50" />
  </svg>
)

export default Circle
