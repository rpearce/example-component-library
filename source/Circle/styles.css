.rl-circle { fill: #686de0; }
