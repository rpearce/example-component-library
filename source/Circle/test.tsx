import React from 'react'
import { render } from '@testing-library/react'
import axe from 'axe-core'
import Circle from './index'

test('with all props', () => {
  const { asFragment } = render(
    <Circle desc="A blue circle" size={200} title="Water planet" />
  )

  expect(asFragment()).toMatchSnapshot()
})

test('without title & desc', () => {
  const { asFragment } = render(<Circle size={200} />)

  expect(asFragment()).toMatchSnapshot()
})

test('is accessible with all props', (done) => {
  const { container } = render(
    <Circle desc="A blue circle" size={200} title="Water planet" />
  )

  axe.run(container, {}, (err, result) => {
    expect(err).toEqual(null)
    expect(result.violations.length).toEqual(0)
    done()
  })
})

test('is inaccessible without title', (done) => {
  const { container } = render(<Circle desc="A blue circle" size={200} />)

  axe.run(container, {}, (err, result) => {
    expect(err).toEqual(null)
    expect(result.violations[0].id).toEqual('svg-img-alt')
    done()
  })
})
