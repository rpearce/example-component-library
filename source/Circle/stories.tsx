import React from 'react'
import { storiesOf } from '@storybook/react'
import { withA11y } from '@storybook/addon-a11y'
import { color, number, text, withKnobs } from '@storybook/addon-knobs'
import '../../.storybook/base.css'
import './styles.css'
import { Circle } from '../../dist/esm'

const stories = storiesOf('Circle', module)

stories.addDecorator(withA11y)
stories.addDecorator(withKnobs)

stories.add('default', () => (
  <Circle
    desc={text('desc', 'A blue circle')}
    fill={color('fill', '#7ed6df')}
    size={number('size', 200)}
    title={text('title', 'Abstract water planet')}
  />
))
