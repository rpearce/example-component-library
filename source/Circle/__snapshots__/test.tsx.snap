// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with all props 1`] = `
<DocumentFragment>
  <svg
    class="rl-circle"
    height="200"
    role="img"
    viewBox="0 0 100 100"
    width="200"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Water planet
    </title>
    <desc>
      A blue circle
    </desc>
    <circle
      cx="50"
      cy="50"
      r="50"
    />
  </svg>
</DocumentFragment>
`;

exports[`without title & desc 1`] = `
<DocumentFragment>
  <svg
    class="rl-circle"
    height="200"
    role="img"
    viewBox="0 0 100 100"
    width="200"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="50"
      cy="50"
      r="50"
    />
  </svg>
</DocumentFragment>
`;
